lib/diff/lcs/ldiff.rb

```diff
@@ -99,74 +108,64 @@ def run(args, _input = $stdin, output = $stdout, error = $stderr) # :nodoc:
     # items we've read from each file will differ by FLD (could be 0).
     file_length_difference = 0
 
-    data_old = File.read(file_old)
-    data_new = File.read(file_new)
-
     # Test binary status
-    if @binary.nil?
-      old_txt = data_old[0, 4096].scan(/\0/).empty?
-      new_txt = data_new[0, 4096].scan(/\0/).empty?
-      @binary = !old_txt || !new_txt
-    end
-
-    unless @binary
-      data_old = data_old.lines.to_a
-      data_new = data_new.lines.to_a
+    if binary.nil?
+      old_txt = info_old.data[0, 4096].scan(/\0/).empty?
+      new_txt = info_new.data[0, 4096].scan(/\0/).empty?
+      binary = !old_txt || !new_txt
```

Comment:
I have merge #104 which will hopefully rebase into this nicely.

Reply:
I got a tiny conflict 3 lines below but yes :)